packages/ERTP/src/purse.js

```diff
@@ -24,41 +29,34 @@ export const makePurseMaker = (allegedName, assetKind, brand, purseMethods) => {
         balanceUpdater,
       };
     },
-    state => {
-      const { balanceNotifier, balanceUpdater } = state;
-      const updatePurseBalance = newPurseBalance => {
-        state.currentBalance = newPurseBalance;
-        balanceUpdater.updateState(state.currentBalance);
-      };
-
+    {
       /** @type {Purse} */
-      const purse = {
-        deposit: (srcPayment, optAmountShape = undefined) => {
+      purse: {
+        deposit: ({ state }, srcPayment, optAmountShape = undefined) => {
```

Comment:
the type of `purse` is no longer `Purse`, which is the use API.

we might want a helper like `WithContext` so that we can still define an external API and then use it here to constrain the implementation. e.g. `/** @type {WithContext<Purse>} */`. Though I'm sure there's a better name.